api/apps/api/src/modules/scenarios-features/split/split-query.service.ts

```diff
@@ -55,62 +55,62 @@ export class SplitQuery {
 
     const hasSubSetFilter = (input.selectSubSets ?? []).length > 0;
     const query = `
-            insert into scenario_features_preparation as sfp (feature_class_id,
-                                                              scenario_id,
-                                                              specification_id,
-                                                              fpf,
-                                                              target,
-                                                              prop,
-                                                              total_area,
-                                                              current_pa)
-            WITH split as (
-              WITH subsets as (
-              select value as sub_value, target, fpf, prop
-              from json_to_recordset('${JSON.stringify(
-                input.selectSubSets ?? [],
-              )}') as x(value varchar, target float8, fpf float8,
-              prop float8)
-              )
-              SELECT distinct fpkv.feature_data_id,
-              fpkv.key,
-              fpkv.value,
-              subsets.fpf,
-              subsets.prop,
-              subsets.target
-              from feature_properties_kv fpkv
-              left join subsets
-              on fpkv.value::TEXT = subsets.sub_value::varchar
-              where fpkv.feature_id = ${fields.featureId}
-              and fpkv.key = ${fields.splitByProperty} ${
+      insert into scenario_features_preparation as sfp (feature_class_id,
+                                                        scenario_id,
+                                                        specification_id,
+                                                        fpf,
+                                                        target,
+                                                        prop,
+                                                        total_area,
+                                                        current_pa)
+      WITH split as (
+        WITH subsets as (
+          select value as sub_value, target, fpf, prop
+          from json_to_recordset('${JSON.stringify(
+            input.selectSubSets ?? [],
+          )}') as x(value varchar, target float8, fpf float8,
+                    prop float8)
+        )
+        SELECT distinct fpkv.feature_data_id,
+                        fpkv.key,
+                        fpkv.value,
+                        subsets.fpf,
+                        subsets.prop,
+                        subsets.target
+        from feature_properties_kv fpkv
+               left join subsets
+                         on trim('"' FROM fpkv.value::TEXT) = trim('"' FROM subsets
+                           .sub_value::text)
```

Comment:
^ The only changed part

Reply:
this is because casting `fpkv.value` to `text` gives us a text representation of the JSON document.

I cannot test quickly right now but if you want to avoid casting and trimming (not sure about the perf penalty, likely marginal if any), IIRC `on fpkv.value->>0` *should* do.

Reply:
@hotzevzl yep, that works for this case but not sure if this will always work (as we have jsonb, we may have something else that primitive types)

![image](https://user-images.githubusercontent.com/3466388/135971496-b128b631-3ef1-4d93-ad50-366d8a1a68f9.png)

Reply:
yep, *in theory* it should work for the features we import via ETL, but without explicit guards to ensure we always have a single string or number (also when importing user-uploaded features), I agree that we can't rely on this assumption.